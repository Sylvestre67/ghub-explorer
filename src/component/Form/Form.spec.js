import React from 'react';
import { shallow } from 'enzyme';
import Button from '@material-ui/core/Button';

import { Form } from './index';
let component;
let instance;

const props = {
	classes: {},
	history: {
		replace: jest.fn(),
		push: jest.fn(),
	},
	location: {
		search: '',
		pathname: 'testing',
	},
};

describe('Form', () => {
	beforeAll(() => {
		component = shallow(<Form {...props} />);
		instance = component.instance();
	});

	it('should render correctly', () => {
		expect(component).toMatchSnapshot();
	});

	describe('Handle form updates', () => {
		it('should update the state with the tested value if not a checkbox', () => {
			instance.handleChange({
				target: { id: 'testing', value: 'tested_value' },
			});
			expect(component.state('testing')).toEqual('tested_value');
		});

		it('should update the state with the boolean if checkbox', () => {
			instance.handleChange({
				target: { id: 'testing', checked: true },
			});
			expect(component.state('testing')).toBeTruthy();
		});
	});

	describe('Click on Submit', () => {
		it('should update the location with the coresponding state values', () => {
			component.find(Button).simulate('click');
			expect(props.history.push).toHaveBeenCalledWith({
				path: 'testing',
				search: 'q=&license=&forked=true&stars=',
			});
		});
	});
});
