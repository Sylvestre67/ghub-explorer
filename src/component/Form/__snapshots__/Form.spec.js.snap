// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should render correctly 1`] = `
<form
  autoComplete="off"
  noValidate={true}
>
  <WithStyles(Grid)
    container={true}
    spacing={16}
  >
    <WithStyles(Grid)
      item={true}
      sm={6}
      xs={12}
    >
      <TextField
        id="q"
        label="Search"
        margin="normal"
        onChange={[Function]}
        required={false}
        select={false}
        value=""
        variant="outlined"
      />
      <TextField
        id="stars"
        label="Stars"
        margin="normal"
        onBlur={[Function]}
        onChange={[Function]}
        required={false}
        select={false}
        value=""
        variant="outlined"
      />
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
      sm={6}
      xs={12}
    >
      <WithStyles(FormControl)
        variant="outlined"
      >
        <WithStyles(WithFormControlContext(InputLabel))
          htmlFor="license"
        >
          License
        </WithStyles(WithFormControlContext(InputLabel))>
        <WithStyles(WithFormControlContext(Select))
          inputProps={
            Object {
              "id": "license",
              "name": "license",
            }
          }
          native={true}
          onChange={[Function]}
          value=""
        >
          <option
            value=""
          />
          <option
            key="mit"
            value="mit"
          >
            MIT
          </option>
          <option
            key="isc"
            value="isc"
          >
            ISC
          </option>
          <option
            key="apache-2.0"
            value="apache-2.0"
          >
            Apache license 2.0
          </option>
          <option
            key="gpl"
            value="gpl"
          >
            GNU General Public License family
          </option>
        </WithStyles(WithFormControlContext(Select))>
      </WithStyles(FormControl)>
      <WithStyles(WithFormControlContext(FormControlLabel))
        control={
          <WithStyles(Checkbox)
            checked={false}
            id="forked"
            onChange={[Function]}
          />
        }
        label="Include Forked"
      />
    </WithStyles(Grid)>
  </WithStyles(Grid)>
  <WithStyles(Grid)
    container={true}
  >
    <WithStyles(Grid)
      item={true}
      xs={12}
    >
      <WithStyles(Button)
        color="primary"
        elevation={0}
        variant="contained"
      >
        Search
      </WithStyles(Button)>
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</form>
`;
